modules/core/src/main/java/org/apache/ignite/internal/managers/communication/GridIoMessage.java

```diff
@@ -67,6 +68,9 @@
     /** Message. */
     private Message msg;
 
+    /** Security subject id that will be used during message processing on an remote node. */
+    private UUID secSubjId;
```

Comment:
This may have pretty significant perofmance impact - we just added 16 bytes to each message. Can you suggest a way to send this field only if security is enabled.

Reply:
Fixed